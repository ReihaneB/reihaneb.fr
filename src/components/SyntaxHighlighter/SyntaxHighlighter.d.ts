export interface SyntaxHighlighterProps {
  language?: string | undefined;
  children: string;
}
