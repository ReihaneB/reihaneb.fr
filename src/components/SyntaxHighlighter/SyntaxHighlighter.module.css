.root {
  border-radius: 16px;
  background-color: var(--blue-lighter);
  overflow: hidden;
}

.language {
  padding: 8px 16px;
  font-size: 12px;
  background-color: var(--blue-light);
  font-weight: 600;
  text-transform: uppercase;
}
