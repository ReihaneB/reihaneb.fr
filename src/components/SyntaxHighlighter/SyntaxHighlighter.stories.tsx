import type { Meta, StoryObj } from '@storybook/react';

import SyntaxHighlighter from './SyntaxHighlighter';

const meta: Meta<typeof SyntaxHighlighter> = {
  title: 'Components/SyntaxHighlighter',
  component: SyntaxHighlighter,
};

export default meta;
type Story = StoryObj<typeof SyntaxHighlighter>;

const defaultArgs = {
  language: 'javascript',
  children: 'console.log("Hello, World!");',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const JavaScript: Story = {
  args: {
    ...defaultArgs,
    children: (
      `function greet(name) {\n return 'Hello, ' + name + '!';\n}\n\nconsole.log(greet('World'));\n`
    ),
  },
  name: 'JavaScript',
};

export const TypeScript: Story = {
  args: {
    children: (
      `function greet(name: string): string {\n return 'Hello, ' + name + '!';\n}\n\nconsole.log(greet('World'));\n`
    ),
    language: 'typescript',
  },
  name: 'TypeScript',
};

export const CSS: Story = {
  args: {
    language: 'css',
    children: (
      `body {\n font-family: 'Inter', sans-serif;\n font-size: 16px;\n}\n`
    ),
  },
};

export const WithouthLanguage: Story = {
  args: {
    children: (
      `body {\n font-family: 'Inter', sans-serif;\n font-size: 16px;\n}\n`
    ),
  },
};
