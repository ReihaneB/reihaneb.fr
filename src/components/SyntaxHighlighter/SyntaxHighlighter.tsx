import { Prism as Highlighter } from 'react-syntax-highlighter';
import { coldarkCold } from 'react-syntax-highlighter/dist/esm/styles/prism';

import { SyntaxHighlighterProps } from './SyntaxHighlighter.d';
import styles from './SyntaxHighlighter.module.css';

function SyntaxHighlighter({
  language,
  children,
  ...rest
}: SyntaxHighlighterProps) {
  return (
    <div className={styles.root}>
      {language && (
        <p className={styles.language}>
          {language}
        </p>
      )}
      <Highlighter
        language={language}
        style={coldarkCold}
        customStyle={{
          backgroundColor: 'transparent',
          margin: 0,
        }}
        codeTagProps={{
          style: {
            backgroundColor: 'transparent',
            fontFamily: '"Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace',
          },
        }}
        {...rest}
      >
        {children}
      </Highlighter>
    </div>
  );
}

export default SyntaxHighlighter;
