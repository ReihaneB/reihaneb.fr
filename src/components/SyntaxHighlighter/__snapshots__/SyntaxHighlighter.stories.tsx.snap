// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/SyntaxHighlighter CSS smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
    <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
      css
    </p>
    <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
      <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; white-space: pre;">
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          body
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          {
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(0, 90, 142);"
        >
          font-family
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          :
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          'Inter'
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ,
        </span>
        <span>
          sans-serif
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(0, 90, 142);"
        >
          font-size
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          :
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(117, 95, 0);"
        >
          16
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          px
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          }
        </span>
        <span>
        </span>
      </code>
    </pre>
  </div>
</main>
`;

exports[`Components/SyntaxHighlighter JavaScript smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
    <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
      javascript
    </p>
    <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
      <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; white-space: pre;">
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          function
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(124, 0, 170);"
        >
          greet
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(0, 90, 142);"
        >
          name
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          {
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token control-flow"
              style="color: rgb(160, 73, 0);"
        >
          return
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          'Hello, '
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          +
        </span>
        <span>
          name
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          +
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          '!'
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          }
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token console"
              style="color: rgb(0, 90, 142);"
        >
          console
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          .
        </span>
        <span class="token method"
              style="color: rgb(0, 90, 142);"
        >
          log
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(124, 0, 170);"
        >
          greet
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          'World'
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
      </code>
    </pre>
  </div>
</main>
`;

exports[`Components/SyntaxHighlighter Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
    <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
      javascript
    </p>
    <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
      <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; white-space: pre;">
        <span class="token console"
              style="color: rgb(0, 90, 142);"
        >
          console
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          .
        </span>
        <span class="token method"
              style="color: rgb(0, 90, 142);"
        >
          log
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          "Hello, World!"
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
      </code>
    </pre>
  </div>
</main>
`;

exports[`Components/SyntaxHighlighter TypeScript smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
    <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
      typescript
    </p>
    <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
      <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; white-space: pre;">
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          function
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(124, 0, 170);"
        >
          greet
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span>
          name
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          :
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(175, 0, 175);"
        >
          string
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          :
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(175, 0, 175);"
        >
          string
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          {
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token control-flow"
              style="color: rgb(160, 73, 0);"
        >
          return
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          'Hello, '
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          +
        </span>
        <span>
          name
        </span>
        <span class="token"
              style="color: rgb(160, 73, 0);"
        >
          +
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          '!'
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          }
        </span>
        <span>
        </span>
        <span>
        </span>
        <span class="token console"
              style="color: rgb(0, 90, 142);"
        >
          console
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          .
        </span>
        <span class="token method"
              style="color: rgb(0, 90, 142);"
        >
          log
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(124, 0, 170);"
        >
          greet
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          (
        </span>
        <span class="token"
              style="color: rgb(17, 107, 0);"
        >
          'World'
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          )
        </span>
        <span class="token"
              style="color: rgb(17, 27, 39);"
        >
          ;
        </span>
        <span>
        </span>
      </code>
    </pre>
  </div>
</main>
`;

exports[`Components/SyntaxHighlighter WithouthLanguage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
    <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
      <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; white-space: pre;">
        <span>
          body {
        </span>
        font-family: 'Inter', sans-serif;
 font-size: 16px;
}
      </code>
    </pre>
  </div>
</main>
`;
